### PR TITLE
Tag DifferentialEquations v0.4.0

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Added dense output via call-overloading, so that way the interpolated solutions are given by `sol(t)` using the same solution object as the timeseries of the output. The new version is also compatible with unit-checked arithmetic via SIUnits.jl (and should be automatically compatible with Unitful.jl, haven't checked it yet). Added a benchmarking suite, and it shows really good results for native DifferentialEquations algorithms over Sundials, ODE, and ODEInterface on nonstiff equations. Extensions to handle explicit parameters for additional features was handled by the creation of ParameterizedFunctions.jl (still experimental). 

Included this time are minimum versions for each dependency.

### CODE REVIEW

File: DifferentialEquations/versions/0.4.0/requires [hunk 1]
```diff
@@ -0,0 +1,10 @@
+julia 0.5-
```
Comment (tkelman) [inline DifferentialEquations/versions/0.4.0/requires]: can wait a while, but when you're ready to drop support for prereleases, take off the trailing dash

**Comment by tkelman** [comment]
you still mention GrowableArrays in `src/precompile.jl` but it's not in REQUIRE any more?

**Comment by tkelman** [comment]
similarly some of the tests use EllipsisNotation, so that should be explicitly in either REQUIRE or test/REQUIRE

**Comment by ChrisRackauckas** [comment]
Those two packages longer used for this (at least for now). The precompilation file is old so it's not being used right now (it's commented out). I plan on updating it after this release, but I'd like to give it a good amount of time on master before I'd put a new one into release. (I'll remove the tests which use EllipsisNotation, I wonder why that wasn't causing test failures?), and fix the tests (had release tests disabled since I dropped v0.4 compatibility, forgot to re-add release for tests yesterday!). I don't want to require v0.5 instead of v0.5- yet because some installations, like the Julia COPR, still have v0.5-rc4 on nightly. I'll post when these changes are made.

**Comment by tkelman** [comment]
One of your other dependencies must indirectly be pulling in EllipsisNotation still. Also if you're going to have `TEST_CONDITIONAL_DEPS` set to true, then `Sundials` should stay in `test/runtests.jl` - people are going to want to run tests locally, and best if `Pkg.test` could work by default.

**Comment by ChrisRackauckas** [comment]
Thanks. I'll work that out.

**Pull request closed without merging**